lib/semian/circuit_breaker_shared_data.rb

```diff
@@ -0,0 +1,70 @@
+module Semian
+  class CircuitBreakerSharedData #:nodoc:
```

Comment:
I'd consider renaming this data structure to `SlidingWindow` and then make this the fallback implementation where all the counts are just in memory for the current VM. If `SysV` is installed, the `SlidingWindow` implementation is overridden by the SysV one, but could also be POSIX, Java, whatever.